crates/api_common/src/utils.rs

```diff
@@ -126,52 +130,91 @@ pub async fn mark_post_as_unread(
     .with_lemmy_type(LemmyErrorType::CouldntMarkPostAsRead)
 }
 
-pub fn check_user_valid(
-  banned: bool,
-  ban_expires: Option<DateTime<Utc>>,
-  deleted: bool,
-) -> Result<(), LemmyError> {
+pub fn check_user_valid(person: &Person) -> Result<(), LemmyError> {
   // Check for a site ban
-  if is_banned(banned, ban_expires) {
+  if person.banned {
```

Comment:
from what I see this changes the meaning of the ban_expires column in the table. before, if banned was true and ban_expires was in the past, it would count as not banned. now it will count as banned. this is intentional?

Reply:
In scheduled_tasks.rs, there's an hourly-run `update_banned_when_expired` task, that should update the banned column, so we should remove any code checks that use the more complicated logic that checks both columns.